tree_test.go

```diff
@@ -1278,3 +1325,90 @@ func BenchmarkBatchLeavesInsert(b *testing.B) {
 		}
 	}
 }
+
+func TestManipulateChildren(t *testing.T) {
```

Comment:
As mentioned, the APIs `Children()` and `SetChild()` aren't used in `go-verkle` anywhere but in iterators in `go-ethereum`. Adding some test cases to cover them.